usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/MeteringEventProcessor.java

```diff
@@ -0,0 +1,74 @@
+package com.sequenceiq.cloudbreak.usage;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.metering.events.MeteringEventsProto;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.sigmadbus.model.DatabusRequest;
+import com.sequenceiq.cloudbreak.sigmadbus.model.DatabusRequestContext;
+import com.sequenceiq.cloudbreak.usage.metering.MeteringDatabusRecordProcessor;
+
+@Service
+public class MeteringEventProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MeteringEventProcessor.class);
+
+    private static final int METERING_EVENT_VERSION = 3;
+
+    private final MeteringDatabusRecordProcessor meteringDatabusRecordProcessor;
+
+    private final RegionAwareInternalCrnGeneratorFactory internalCrnGeneratorFactory;
+
+    public MeteringEventProcessor(MeteringDatabusRecordProcessor meteringDatabusRecordProcessor,
+            RegionAwareInternalCrnGeneratorFactory internalCrnGeneratorFactory) {
+        this.meteringDatabusRecordProcessor = meteringDatabusRecordProcessor;
+        this.internalCrnGeneratorFactory = internalCrnGeneratorFactory;
+    }
+
+    public void processEvent(MeteringEventsProto.MeteringEvent event) {
+        DatabusRequestContext dbusContext = DatabusRequestContext.Builder.newBuilder()
+                .withAccountId(getAccountId())
+                .build();
+        DatabusRequest databusRequest = DatabusRequest.Builder.newBuilder()
+                .withMessageBody(event)
+                .withContext(dbusContext)
+                .build();
+        meteringDatabusRecordProcessor.processRecord(databusRequest);
+    }
+
+    public void storageHeartbeat(MeteringEventsProto.StorageHeartbeat details, MeteringEventsProto.ServiceType.Value serviceType) {
+        try {
+            checkNotNull(details);
+            checkNotNull(serviceType);
+            MeteringEventsProto.MeteringEvent event = eventBuilder()
+                    .setStorageHeartbeat(details)
+                    .setServiceType(serviceType)
+                    .setServiceConfiguration(MeteringEventsProto.ServiceConfiguration.newBuilder()
+                            .setServiceFeature(MeteringEventsProto.ServiceFeature.Value.OBJECT_STORAGE).build())
+                    .build();
+            processEvent(event);
+            LOGGER.info("Sent binary format for the following metering event: {}", details);
+        } catch (Exception e) {
+            LOGGER.warn("Could send binary format for the following metering event: {}! Cause: {}", details, e.getMessage());
+        }
```

Comment:
* It would be nice to log the whole exception, not only its message.
* Is it okay to otherwise ignore the exception, i.e. not propagate it back to the caller (as if the sending was successful)?

Reply:
Good catch, it's indeed better to throw the exception after logging so the storage consumption flow can fail in this case and we'll have an indication that something went wrong.